referenceframe/worldstate.go

```diff
@@ -71,18 +73,29 @@ func WorldStateToProtobuf(worldState *WorldState) (*commonpb.WorldState, error)
 // in the WorldState such that they are in the frame system's World reference frame.
 func (ws *WorldState) ToWorldFrame(fs FrameSystem, inputs map[string][]Input) (*WorldState, error) {
 	transformGeometriesToWorldFrame := func(gfs []*GeometriesInFrame) (*GeometriesInFrame, error) {
-		allGeometries := make(map[string]spatial.Geometry)
-		for name1, gf := range gfs {
+		nameCheck := make(map[string]bool)
+		allGeometries := make([]spatial.Geometry, 0, len(gfs))
+
+		unnamedCount := 1
+
+		for _, gf := range gfs {
 			tf, err := fs.Transform(inputs, gf, World)
 			if err != nil {
 				return nil, err
 			}
-			for name2, g := range tf.(*GeometriesInFrame).Geometries() {
-				geomName := strconv.Itoa(name1) + "_" + name2
-				if _, present := allGeometries[geomName]; present {
-					return nil, errors.New("multiple geometries with the same name")
+			for _, g := range tf.(*GeometriesInFrame).Geometries() {
+				geomName := g.Label()
+				if geomName == "" {
+					geomName = defaultWorldStateGeomPrefix + strconv.Itoa(unnamedCount)
+					g.SetLabel(geomName)
+					unnamedCount++
+				}
+
+				if _, present := nameCheck[geomName]; present {
+					return nil, fmt.Errorf("cannot specify multiple geometries with the same name: %s", geomName)
```

Comment:
Why are we checking for names in the transform function and not the instantiation of the worldState itself?